lib/vagrant-hosts/cap/sync_hosts/windows.rb

```diff
@@ -19,4 +19,65 @@ def update_hosts
     @machine.communicate.sudo(script.join("; "))
   end
 
+  # Windows needs a modification of the base method because Windows guest names
+  # cannot be fully-qualified domain names (they cannot contain the "."
+  # character). Therefore, modify the input name to convert illegal characters
+  # to legal replacements.
+  #
+  # @param name [String] The new hostname to apply on the guest
+  def change_host_name(name)
+
+    # First set the machine name (hostname)
+    components = name.split('.')
+    hostname   = components.first
+    domainname = components.slice(1, components.size).join('.')
+
+    super(hostname)
+
+    # Next set the Primary DNS Suffix, if it makes sense (domainname)
+    unless domainname.empty?
+      change_domain_name(domainname)
+    end
+  end
+
+  def change_domain_name(domainname)
+    # Source: http://poshcode.org/2958
+    # Note that whitespace is important in this inline powershell script due
+    # to the use of a here-string.
+    powershell = <<-END_OF_POWERSHELL
+function Set-PrimaryDnsSuffix {
+  param ([string] $Suffix)
+
+  # http://msdn.microsoft.com/en-us/library/ms724224(v=vs.85).aspx
+  $ComputerNamePhysicalDnsDomain = 6
+
+  Add-Type -TypeDefinition @"
+  using System;
+  using System.Runtime.InteropServices;
+
+  namespace ComputerSystem {
+      public class Identification {
+          [DllImport("kernel32.dll", CharSet = CharSet.Auto)]
+          static extern bool SetComputerNameEx(int NameType, string lpBuffer);
```

Comment:
In exploring this it also seemed like the Workgroup wasn't quite the same thing as what on Linux would be the domainname, which seems to map most closely to the Primary DNS Suffix on Windows. Especially if Workgroups can't have more than 15 characters in them, changing the Workgroup might not get us what we need in this case.